tests/app/phase-team-response/issues-faulty/issues-faulty.component.spec.ts

```diff
@@ -0,0 +1,55 @@
+import { IssuesFaultyComponent } from '../../../../src/app/phase-team-response/issues-faulty/issues-faulty.component';
+import { Issue, STATUS } from '../../../../src/app/core/models/issue.model';
+import { ISSUE_WITH_EMPTY_DESCRIPTION } from '../../../constants/githubissue.constants';
+import { Team } from '../../../../src/app/core/models/team.model';
+import { IssueService } from '../../../../src/app/core/services/issue.service';
+import { UserService } from '../../../../src/app/core/services/user.service';
+import { USER_Q, TEAM_4 } from '../../../constants/data.constants';
+
+describe('IssuesFaultyComponent', () => {
+  describe('.ngOnInit()', () => {
+    const dummyTeam: Team = TEAM_4;
+    let dummyIssue: Issue;
+    let issuesFaultyComponent: IssuesFaultyComponent;
+    const issueService: IssueService = new IssueService(null, null, null, null, null, null, null);
+    const userService: UserService = new UserService(null, null);
+    userService.currentUser = USER_Q;
+    const DUMMY_DUPLICATE_ISSUE_ID = 1;
+    const DUMMY_RESPONSE = 'dummy response';
+
+    beforeEach(() => {
+      dummyIssue = Issue.createPhaseTeamResponseIssue(ISSUE_WITH_EMPTY_DESCRIPTION, dummyTeam);
+      issueService.updateLocalStore(dummyIssue);
+      issuesFaultyComponent = new IssuesFaultyComponent(issueService, userService, null);
+      issuesFaultyComponent.ngOnInit();
+    });
+
+    it('should set filter to return false for an issue with no team response', () => {
+      issueService.issues[dummyIssue.id].teamResponse = undefined;
+
+      expect(issuesFaultyComponent.filter(dummyIssue)).toBeFalse();
+    });
+
+    it('should set filter to return false for a non-duplicate issue with responses', () => {
+      issueService.issues[dummyIssue.id].teamResponse = DUMMY_RESPONSE;
+      dummyIssue.duplicateOf = undefined;
+
+      expect(issuesFaultyComponent.filter(dummyIssue)).toBeFalse();
+    });
+
+    it('should set filter to return false for a duplicate issue with responses that is not duplicated by other issues', () => {
+      issueService.issues[dummyIssue.id].teamResponse = DUMMY_RESPONSE;
+      dummyIssue.duplicateOf = DUMMY_DUPLICATE_ISSUE_ID;
+
+      expect(issuesFaultyComponent.filter(dummyIssue)).toBeFalse();
```

Comment:
Actually since the `dummyIssue` is updated by reference maybe we can just standardize it to this? Across the tests?

```suggestion
      issueService.issues[dummyIssue.id].teamResponse = DUMMY_RESPONSE;
      issueService.issues[dummyIssue.id].duplicateOf = DUMMY_DUPLICATE_ISSUE_ID;

      expect(issuesFaultyComponent.filter(dummyIssue)).toBeFalse();
```

or something like this?

```suggestion
      const testIssue: Issue = {
        ...dummyIssue, 
        teamResponse: DUMMY_RESPONSE,
        duplicateOf: DUMMY_DUPLICATE_ISSUE_ID
      } as Issue;
      issueService.updateLocalStore(testIssue);

      expect(issuesFaultyComponent.filter(testIssue)).toBeFalse();
```

What do you think? Its more of a code style thing rather than functionality.

Reply:
destructuring looks neat! i'll do that for the testcases